### PR TITLE
Add multiple-org setup email for start of cycle notifications

### PR DESCRIPTION
**Pull request by steventux**
## Context

We have 2 different designs for users who need to set up organisation permissions, one for users who belong to a single organisation which has permissions to set up and another for users who belong to multiple organisations with permissions to set up.

## Changes proposed in this pull request

Groups the pending relationships into a hash keyed by provider names which the user belongs to, and values which are the partner organisations which relate to the set up.
 

## Link to Trello card

https://trello.com/c/j6AXBzwy/4291-create-start-of-cycle-provider-emails-for-find-and-apply

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: spec/mailers/previews/provider_mailer_preview.rb [hunk 1]
```diff
@@ -120,10 +120,12 @@ def find_service_is_now_open
   end
 
   def set_up_organisation_permissions
```
Comment (cwrw) [inline spec/mailers/previews/provider_mailer_preview.rb:122]: maybe we can have two views to show both variations?

File: app/views/provider_mailer/set_up_organisation_permissions.text.erb [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+<% if @multiple_relationships -%>
+Candidates can now find courses on GOV.UK that you work on with the partner organisations listed below.
+
+You cannot manage applications to these courses until either you or your partner organisations have set up organisational permissions.
+
+<% @relationships_to_set_up.each do |organisation, partner_organisations| -%>
+
+For <%= organisation %>, you need to set up permissions for courses you work on with:
+
+<% partner_organisations.each do |partner_organisation| -%>
```
Comment (cwrw) [inline app/views/provider_mailer/set_up_organisation_permissions.text.erb:12]: could we fix the indentation here?

File: spec/mailers/previews/provider_mailer_preview.rb (hunk 1 above)
Comment (steventux) [reply spec/mailers/previews/provider_mailer_preview.rb:122]: Yeah I agree it's a bit messy atm

File: app/views/provider_mailer/set_up_organisation_permissions.text.erb (hunk 2 above)
Comment (steventux) [reply app/views/provider_mailer/set_up_organisation_permissions.text.erb]: The issue I found when doing this was that the markdown lists do not render correctly if indented. https://github.com/DFE-Digital/apply-for-teacher-training/pull/5676/commits/4a477dc867d32eda76d91e67538c881fe0f47ba0 simplifies the conditionals a bit.

File: spec/mailers/previews/provider_mailer_preview.rb (hunk 1 above)
Comment (steventux) [reply spec/mailers/previews/provider_mailer_preview.rb:122]: https://github.com/DFE-Digital/apply-for-teacher-training/commit/4a477dc867d32eda76d91e67538c881fe0f47ba0

**Review by cwrw — APPROVED** [review]

**Pull request merged**